src/coreclr/jit/async.cpp

```diff
@@ -956,18 +958,25 @@ ContinuationLayout AsyncTransformation::LayOutContinuation(BasicBlock*
         if (dsc->TypeIs(TYP_STRUCT) || dsc->IsImplicitByRef())
         {
             ClassLayout* layout = dsc->GetLayout();
-            assert(!layout->HasGCByRef());
 
-            if (layout->IsCustomLayout())
+            // TODO: (async) we do not need to save ByRef-containing locals
+            //       as by the spec an await turns them into zero-inited state.
+            //       For now just store/restore as if there are no gc refs.
+            //       This is mostly to handle the "fake" live-across-await byrefs
+            //       in min-opts, since C#-compiled code by itself does not let
+            //       byrefs be live across awaits.
+            unsigned objCount = layout->HasGCByRef() ? 0 : layout->GetGCPtrCount();
```

Comment:
We will need to figure out what is creating these LIR edges that are live across the await and stop doing that. This fix will replace assert with bad codegen instead.